src/utils/releases.js

```diff
@@ -56,7 +56,35 @@ async function generateReleaseNotes(token, newVersion, baseVersion) {
   }
 }
 
+async function fetchReleaseByTag(token, tag) {
```

Comment:
Nit: Maybe I'm missing something but it seems you're not actually using the return but rather checking if it doesn't throw. Maybe it could not return and be named something like `ensureReleaseTagExists` so the intention is more clear